### PR TITLE
SIFT custom

### PR DESCRIPTION
**Pull request by carpino**
Hello, 

ive crearted a procedure to instantiate a SIFT object with custom values, 

ive created a new procedure leaving untouched the default unparametrized procedure,

hope this will be included in the dev branch,

(ive closed the pull request just created in the wrong branch)

@deadprogram

### CODE REVIEW

File: features2d.go [hunk 1]
```diff
@@ -943,6 +943,39 @@ func NewSIFT() SIFT {
 	return SIFT{p: unsafe.Pointer(C.SIFT_Create())}
 }
 
+func NewSIFTCustom(nfeatures *int, nOctaveLayers *int, contrastThreshold *float64, edgeThreshold *float64, sigma *float64) SIFT {
```
Comment (deadprogram) [inline features2d.go]:
```suggestion
func NewSIFTWithParams(nfeatures *int, nOctaveLayers *int, contrastThreshold *float64, edgeThreshold *float64, sigma *float64) SIFT {
```

File: features2d.go [hunk 2]
```diff
@@ -943,6 +943,39 @@ func NewSIFT() SIFT {
...
+		numEdgeThreshold = *edgeThreshold
+	}
+
+	var numSigma float64 = 1.6
+
+	if sigma != nil {
+		numSigma = *sigma
+	}
+
+	return SIFT{p: unsafe.Pointer(C.SIFT_CreateCustom(C.int(numFeatures), C.int(numOctaveLayers), C.double(numContrastThreshold), C.double(numEdgeThreshold), C.double(numSigma)))}
```
Comment (deadprogram) [inline features2d.go]:
```suggestion
	return SIFT{p: unsafe.Pointer(C.SIFT_CreateWithParams(C.int(numFeatures), C.int(numOctaveLayers), C.double(numContrastThreshold), C.double(numEdgeThreshold), C.double(numSigma)))}
```

File: features2d.h [hunk 3]
```diff
@@ -97,6 +97,7 @@ struct MultiDMatches FlannBasedMatcher_KnnMatch(FlannBasedMatcher f, Mat query,
 void DrawKeyPoints(Mat src, struct KeyPoints kp, Mat dst, const Scalar s, int flags);
 
 SIFT SIFT_Create();
+SIFT SIFT_CreateCustom(int nfeatures, int nOctaveLayers, double contrastThreshold, double edgeThreshold, double sigma);
```
Comment (deadprogram) [inline features2d.h]:
```suggestion
SIFT SIFT_CreateWithParams(int nfeatures, int nOctaveLayers, double contrastThreshold, double edgeThreshold, double sigma);
```

File: features2d.cpp [hunk 4]
```diff
@@ -590,6 +590,11 @@ SIFT SIFT_Create() {
     return new cv::Ptr<cv::SIFT>(cv::SIFT::create());
 }
 
+SIFT SIFT_CreateCustom(int nfeatures, int nOctaveLayers, double contrastThreshold, double edgeThreshold, double sigma) {
```
Comment (deadprogram) [inline features2d.cpp]:
```suggestion
SIFT SIFT_CreateWithParams(int nfeatures, int nOctaveLayers, double contrastThreshold, double edgeThreshold, double sigma) {
```

**Comment by deadprogram** [comment]
Hello @carpino please see my suggested changes inline.

See https://github.com/hybridgroup/gocv/blob/release/CONTRIBUTING.md#how-to-add-a-function-from-opencv-to-gocv for some explanation.

Also, can you please add a smoke test as described in the contribution guidelines?

Thank you!

**Comment by carpino** [comment]
> Hello @carpino please see my suggested changes inline.
> 
> See https://github.com/hybridgroup/gocv/blob/release/CONTRIBUTING.md#how-to-add-a-function-from-opencv-to-gocv for some explanation.
> 
> Also, can you please add a smoke test as described in the contribution guidelines?
> 
> Thank you!

Hi @deadprogram, thanks for the quick response, changed the name according to your suggestion and contribution guidelines,

added smoke test that follow the structure of the test for SIFT without params, checking the correct number of features, generally I should check for a number of features not greater than the params but for the image used in the test SIFT extract an higher number of features if created without params, so I check an inequality

**Comment by deadprogram** [comment]
Thanks for the addition @carpino now squash/merging.

**Pull request merged**